### PR TITLE
Usability/clean deps

### PR DESCRIPTION
**Pull request by thorrester**
### Short Summary
Cleans up dependencies and usability

### Context
Version 2 aims to simplify the user experience by installing minimal dependencies when working via a client.

Install options will now be:

Types of installs:
  - `opsml`
       Client dependencies only
  - `opsml[server]`
       client + server deps (sqlalchemy, fastapi, etc)
  - `opsml[sklearn_onnx]`
       skl2onnx + onnx deps
  - `opsml[torch_onnx]`
       onnx deps for pytorch
 - ` opsml[tf_onnx]`
       tf2onnx + onnx deps

### Is this a Breaking Change?
Yes - `V2`

### CODE REVIEW

File: opsml/model/types.py [hunk 1]
```diff
@@ -297,3 +297,36 @@ class ModelDownloadInfo(BaseModel):
     version: Optional[str] = None
     team: Optional[str] = None
     uid: Optional[str] = None
+
+
+class BaseEstimator(Protocol):
```
Comment (damonallison) [inline opsml/model/types.py:302]: What is this blank protocol for? If it doesn't have any methods do we need it?

File: pyproject.toml [hunk 2]
```diff
@@ -75,14 +67,27 @@ mlflow = ">=2.3.1, <=2.4"
 # optional dependencies for sql
 psycopg2 = { version = "^2.9.6", optional = true }
 
+# optional dependencies for onnx
```
Comment (damonallison) [inline pyproject.toml:70]: I'm fine with making these optional by default. We should make sure the default for `to_onnx` is `False` if we are going with onnx off by default or `True` if we decide to make onnx on by default.

**Review by damonallison — DISMISSED** [review]
👍🏻

File: opsml/model/types.py (hunk 1 above)
Comment (thorrester) [reply opsml/model/types.py:302]: It's used as a type hint in `model_converters.py` to show an Sklearn BaseEstimator is expected without actually import sklearn

**Pull request merged**